core/templates/dev/head/i18n.js

```diff
@@ -83,13 +104,24 @@ oppia.config([
     .translations('en', DEFAULT_TRANSLATIONS)
     .fallbackLanguage('en')
     .determinePreferredLanguage()
-    .useCookieStorage()
+    .useStorage('customCookieStorage')
     // The strategy 'sanitize' does not support utf-8 encoding.
     // https://github.com/angular-translate/angular-translate/issues/1131
     // The strategy 'escape' will brake strings with raw html, like hyperlinks
     .useSanitizeValueStrategy('sanitizeParameters')
     .forceAsyncReload(true);
-}]);
+}]).run(function($translate, $window) {
+  // Set the site language in case the exploration is embedded in an iframe.
```

Comment:
It's possible that some of this might not be needed, if we use Oppia's built-in iframe detection.